### PR TITLE
PB-7463: Pre-upgrade hook changes for MongoDB Multi-Upgrade

### PR DESCRIPTION
**Pull request by ss-px**
**What this PR does / why we need it**: Pre-upgrade hook changes for MongoDB Multi-Upgrade

**Which issue(s) this PR fixes** (optional)
Closes #619 

**Special notes for your reviewer**: https://portworx.testrail.net/index.php?/runs/view/10081&group_by=cases:section_id&group_order=asc

### CODE REVIEW

**Review by t-nidhin — APPROVED** [review]

File: charts/px-central/templates/px-backup/pre-upgrade-hook/pre-upgrade-check.yaml [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+                values:
+                - "false"
+              {{- end }}
+      containers:
+        - name: pxbackup-mongodb-upgrade
+          image: {{ printf "%s/%s/%s:%s" .Values.images.preUpgradeHookImage.registry .Values.images.preUpgradeHookImage.repo .Values.images.preUpgradeHookImage.imageName .Values.images.preUpgradeHookImage.tag }}
+          imagePullPolicy: {{ .Values.images.pullPolicy }}
+          command: ["python", "/pxcentral-pre-upgrade/pxc-pre-upgrade.py"]
+          env:
+            - name: STATEFULSET_NAME
```
Comment (px-kesavan) [inline charts/px-central/templates/px-backup/pre-upgrade-hook/pre-upgrade-check.yaml:139]: lets be more specific on naming ,  MONGODB_STATEFULSET_NAME

File: charts/px-central/templates/px-backup/pre-upgrade-hook/pre-upgrade-check.yaml [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+              value: pxc-backup-mongodb
+            - name: NAMESPACE
+              value: {{ .Release.Namespace }}  
+            - name: TARGET_VERSION
```
Comment (px-kesavan) [inline charts/px-central/templates/px-backup/pre-upgrade-hook/pre-upgrade-check.yaml:143]:
```suggestion
            - name: MONGODB_TARGET_VERSION
```

File: charts/px-central/templates/px-backup/pre-upgrade-hook/pre-upgrade-check.yaml [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+    name: pxcentral-pre-upgrade-hook
+    app.kubernetes.io/component: pxcentral-pre-upgrade-hook
+  annotations:
+    "helm.sh/hook": pre-upgrade
+    "helm.sh/hook-weight": "2"
+    "helm.sh/hook-delete-policy": before-hook-creation,hook-succeeded,hook-completed
+    meta.helm.sh/release-name: {{ .Release.Name }}
+    meta.helm.sh/release-namespace: {{ .Release.Namespace }}
+spec:
+  template:
```
Comment (px-kesavan) [inline charts/px-central/templates/px-backup/pre-upgrade-hook/pre-upgrade-check.yaml:112]: Add annotation for azure proxy
```suggestion
  template:
    {{- if eq $azureProxyEnabled true }}
    metadata:
      annotations:
        kubernetes.azure.com/no-http-proxy-vars: "true"
    {{- end }}
```

File: charts/px-central/templates/px-backup/pre-upgrade-hook/pre-upgrade-check.yaml [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+                operator: NotIn
+                values:
+                - "false"
+              {{- end }}
+      containers:
+        - name: pxbackup-mongodb-upgrade
+          image: {{ printf "%s/%s/%s:%s" .Values.images.preUpgradeHookImage.registry .Values.images.preUpgradeHookImage.repo .Values.images.preUpgradeHookImage.imageName .Values.images.preUpgradeHookImage.tag }}
+          imagePullPolicy: {{ .Values.images.pullPolicy }}
+          command: ["python", "/pxcentral-pre-upgrade/pxc-pre-upgrade.py"]
+          env:
```
Comment (px-kesavan) [inline charts/px-central/templates/px-backup/pre-upgrade-hook/pre-upgrade-check.yaml:138]: Add env for NO_Proxy support
```suggestion
          env:
          {{- include "proxy.noProxyEnv" . | nindent 10 }}
```

File: charts/px-central/templates/px-backup/pre-upgrade-hook/pre-upgrade-check.yaml [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+  annotations:
+    "helm.sh/hook": pre-upgrade
+    "helm.sh/hook-weight": "0"
+    "helm.sh/resource-policy": keep
+spec:
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 1Gi
```
Comment (px-kesavan) [inline charts/px-central/templates/px-backup/pre-upgrade-hook/pre-upgrade-check.yaml:94]: will 1Gi be sufficient for mongo and mysql dump ?

Comment (ss-px) [reply charts/px-central/templates/px-backup/pre-upgrade-hook/pre-upgrade-check.yaml:94]: increasing it to 170Gi, now that we're using a single PVC to store dumps from both DBs along with the logs.

Comment (ss-px) [reply charts/px-central/templates/px-backup/pre-upgrade-hook/pre-upgrade-check.yaml]:
```
pxc-mongodb-data-pxc-backup-mongodb-0                                                      Bound    pvc-2fab2228-66fb-48a6-983a-96fac7ac140b   64Gi       RWO            portworx-sc    <unset>                 2d12h
pxcentral-mysql-pvc                                                                        Bound    pvc-3a8ef78c-eafe-40b4-970f-7be1702229cd   100Gi      RWO            portworx-sc    <unset>                 2d12h
```
For a MongoDB PVC of 64Gi and a MySQL PVC of 100Gi, plus some margin, the estimated dump size is around 170Gi.

**Pull request closed without merging**